prog/dftb+/lib_dftbplus/parser.F90

```diff
@@ -3667,46 +3662,64 @@ end subroutine readOptions
 
 
   !> Reads in dispersion related settings
-  subroutine readDispersion(node, geo, input, nrChrg)
+  subroutine readDispersion(node, geo, ctrl)
```

Comment:
Please revert this one, if you need additional components from the control structure, pass them explicitly. The dispersion parser should not know about the complete control structure, only about its own input structure.

Reply:
Happy to do so, but this change was done by @bhourahine, so you may want to agree on this first.

Reply:
Sure, revert.

Reply:
Hm, actually after this refactor, `readDispersion()` uses also `ctrl%tSCC` and `ctrl%isSccConvRequired`. Should this be both just explicitly in?

Reply:
Yeah, passing them explicitly it is still better than having the whole control structure there, it allows to reuse the parser at a different position in the tree, even if this is rather unlikely for dispersion. And six dummy arguments is still okay, we have subroutines with over 20 around.